partition_snapshot_reader.hh

```diff
@@ -164,7 +165,8 @@ class partition_snapshot_flat_reader : public flat_mutation_reader::impl, public
         explicit lsa_partition_reader(const schema& s, reader_permit permit, partition_snapshot_ptr snp,
                                       logalloc::region& region, logalloc::allocating_section& read_section,
                                       bool digest_requested)
-            : _schema(s)
+            : _query_schema(s)
+            , _snapshot_schema(s.shared_from_this())
```

Comment:
I wonder if `_snapshot_schema` is the right name...
If I understand correctly, it's really the query schema but maybe unreversed (in later commits). "Query schema but modified to use the snapshot order". This does not necessarily have to be exactly the snapshot schema even though the ordering is the same: the query schema may use a newer version.

Perhaps finding the perfect name does not make much sense, but a comment at where `_snapshot_schema` is defined would be nice. The comment should explain what `_snapshot_schema` "really" is.